From 758de116d59380f1535d93310147b61ae3b7c473 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Mon, 17 Feb 2014 10:29:09 +0800
Subject: [PATCH 493/635] ENGR00214328 fix fsl gralloc register buffer error.

Buffer handle in different process has its own virtual address.
but buffer handle transfered to another process copy the original
pid, so it will generate error when the buffer handle transfer back
to the original process. Because it will use another process's
virtual address due to the same pid.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx6/libgralloc_wrapper/fsl_mapper.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mx6/libgralloc_wrapper/fsl_mapper.cpp b/mx6/libgralloc_wrapper/fsl_mapper.cpp
index 7c39fe1..5b694f2 100644
--- a/mx6/libgralloc_wrapper/fsl_mapper.cpp
+++ b/mx6/libgralloc_wrapper/fsl_mapper.cpp
@@ -97,6 +97,7 @@ int fsl_gralloc_register_buffer(gralloc_module_t const* module,
     if (hnd->pid != getpid()) {
         void *vaddr;
         ret = fsl_gralloc_map(module, handle, &vaddr);
+        hnd->pid = getpid();
     }
     if (ret == 0) {
         // increase the buffer counter in process.
-- 
1.8.0

